File: lab_6/LabOop/Rectangle.cs
using System;

namespace lab_6.LabOop {
    public class Rectangle : GeometricFigure {
        private double width;
        private double height;
        private Color color;

        public override string FigureName { get; } = "Прямоугольник";

        public Rectangle(double width, double height, string color) {
            if (width <= 0 || height <= 0)
                throw new ArgumentException("Ширина и высота должны быть положительными числами");

            this.width = width;
            this.height = height;
            this.color = new Color(color);
        }

        public double Width {
            get { return width; }
        }

        public double Height {
            get { return height; }
        }

        public string FigureColor {
            get { return color.ColorName; }
        }

        public override double CalculateArea() {
            return width * height;
        }

        public override string Repr() {
            return $"Фигура: {FigureName}, ширина: {width:F2}, высота: {height:F2}, цвет: {color}, площадь: {CalculateArea():F2}";
        }

        public override string ToString() {
            return Repr();
        }
    }
}

File: lab_8/MyOop/Rectangle.cpp
#include "Rectangle.h"

// вот это стат поле не забыть
const std::string Rectangle::figureName = "Прямоугольник";

Rectangle::Rectangle(double width, double height, const std::string& color) 
    : width(width), height(height), color(color) {
    if (width <= 0 || height <= 0) {
        throw std::invalid_argument("Ширина и высота !< 0");
    }
}

double Rectangle::area() const {
    return width * height;
}

std::string Rectangle::getFigureName() const {
    return figureName;
}

// вывод не забыть
std::string Rectangle::repr() const {
    std::ostringstream oss;
    oss << std::fixed << std::setprecision(2);
    oss << "Фигура: " << figureName 
        << ", ширина: " << width 
        << ", высота: " << height 
        << ", цвет: " << color.getColor() 
        << ", площадь: " << area();
    return oss.str();
}